gremlin-dotnet/test/Gremlin.Net.UnitTest/Process/Traversal/TraversalTests.cs

```diff
@@ -98,6 +98,17 @@ public void ShouldDrainAllTraversersWhenIterateIsCalled()
 
             Assert.Null(drainedTraversal.Next());
         }
+        
+        [Fact]
+        public void ShouldReturnNullWhenTraverserIsEmpty()
+        {
+            var someObjs = new List<object?>();
+            var traversal = new TestTraversal(someObjs);
+
+            var emptyTraversal = traversal.Iterate();
+
+            Assert.Null(emptyTraversal.Next());
```

Comment:
I'm not sure if returning `null` is the more proper behavior. `Iterate()` is called on the Traversal which will move to the end of the enumerator. Then `Next()` is called which I think would be reasonable to throw an exception.